cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/source/FileBatchSource.java

```diff
@@ -133,13 +134,19 @@ public void prepareRun(BatchSourceContext context) throws Exception {
 
     if (config.timeTable != null) {
       table = context.getDataset(config.timeTable);
-      String lastTimeRead = Bytes.toString(table.read(LAST_TIME_READ));
-      if (lastTimeRead == null) {
-        lastTimeRead = "0";
+      String datesToRead = Bytes.toString(table.read(LAST_TIME_READ));
+      if (datesToRead == null) {
+        ArrayList<Date> firstRun = new ArrayList();
+        firstRun.add(new Date(0));
```

Comment:
Also, you can merge this with the last statement.

``List<Date> firstRun = Lists.newArrayList(new Date(0));``